hydra-node/src/Hydra/Ledger/Cardano/Json.hs

```diff
@@ -471,17 +471,52 @@ instance
   ( FromJSON (Core.TxBody era)
   , FromJSON (Core.AuxiliaryData era)
   , FromJSON (Core.Script era)
+  , FromCBOR (Annotator (Core.TxBody era))
+  , FromCBOR (Annotator (Core.AuxiliaryData era))
+  , FromCBOR (Annotator (Core.Witnesses era))
   , Core.Script era ~ Ledger.Alonzo.Script era
+  , Ledger.ValidateScript era
   , Era era
   ) =>
   FromJSON (Ledger.Alonzo.ValidatedTx era)
   where
-  parseJSON = withObject "Tx" $ \o ->
-    Ledger.Alonzo.ValidatedTx
-      <$> o .: "body"
-      <*> o .: "witnesses"
-      <*> o .:? "isValid" .!= Ledger.Alonzo.IsValid True
-      <*> o .:? "auxiliaryData" .!= SNothing
+  parseJSON value =
+    -- We accepts transactions in three forms:
+    --
+    -- (a) As high-level JSON object, which full format is specified via a
+    -- JSON-schema.
+    --
+    -- (b) As a JSON 'text-envelope', which is a format defined and produced by
+    -- the cardano-cli, wrapping base16-encoded strings as JSON objects with
+    -- tags.
+    --
+    -- (c) As base16 string representing a CBOR-serialized transaction, since
+    -- this is the most common medium of exchange used for transactions.
+    parseAsJSONObject value <|> parseAsBase16CBOR value
```

Comment:
On second-thought, this would be better written as two alternatives between three possibilities, instead of doing two of them within `parseAsJSONObject`; And we may want to put the raw base16-CBOR first as it's likely going to be the most commonly used.

Reply:
Would prefer the three alternatives as three parse functions as well, yes. The documentation already lends itself to be split between these three functions :)